service/src/main/java/org/mskcc/cmo/metadb/service/impl/MetaDbRequestServiceImpl.java

```diff
@@ -120,4 +129,42 @@ private Boolean timestampWithin24Hours(Date referenceTimestamp, Date newTimestam
         adjustedReferenceTimestamp.add(Calendar.MILLISECOND, TIME_ADJ_24HOURS_MS);
         return newTimestamp.before(adjustedReferenceTimestamp.getTime());
     }
+    
+    private RequestMetadata extractRequestMetadata(String requestMetadataJson)
+            throws JsonMappingException, JsonProcessingException {
+        Map<String, String> requestMetadataMap = mapper.readValue(requestMetadataJson.toString(), Map.class);
+        requestMetadataMap.remove("samples");
+        RequestMetadata requestMetadata = new RequestMetadata(
+                mapper.convertValue(requestMetadataMap, String.class),
+                LocalDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE));
+        return requestMetadata;
+    }
+    
+    private String updateRequestJson(String requestMetadataJson, String currentRequestJson)
+            throws JsonMappingException, JsonProcessingException {
+        final ObjectMapper mapper = new ObjectMapper();
+        Map<String, String> currentRequestMap = mapper.readValue(currentRequestJson.toString(), Map.class);
+        Map<String, String> updatedRequestMap = mapper.readValue(requestMetadataJson.toString(), Map.class);
+         
+        for (Map.Entry<String,String> entry : updatedRequestMap.entrySet()) {
+            currentRequestMap.replace(entry.getKey(), entry.getValue());
+        }
+        
+        return mapper.writeValueAsString(currentRequestJson);
+    }
+    
+    private void updateRequestMetadata(String requestMetadataJson)
```

Comment:
if I understand correctly then this method would be called if we already have an existing request in the database? So it would probably be called [in this else block](https://github.com/mskcc/cmo-metadb/pull/364/files#diff-2fee701fd9cface3562b4c12c46a0c818112e6f5d522fd30719778d703b54369R77)

Maybe the `MetadbRequest.addRequestMetadata()` should not just append the list but also call its other method `updateRequestMetadata()` internally so that in one step you are 
1. updating the requestMetadataList
2. updating the `MetadbRequest` data members
3. maybe also update the value of `MetadbRequest.requestJson` (the json string?)

I think that would allow us to get rid of this method here. It Also would allow us to change the inputs of `metadbRequest.updateRequestMetadata()` since `metadbRequest` object would already have access to its current request json string

Thoughts?

If we move forward with this then I wouldn't worry about implementing these changes for this PR specifically. We can have it in a separate PR.